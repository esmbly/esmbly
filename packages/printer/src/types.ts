export type Callback = (error: Error | null | undefined) => void;

export type Encoding = string | undefined;

export type WriteFn = (
  chunk: string,
  encoding?: Encoding,
  callback?: Callback,
) => boolean;
