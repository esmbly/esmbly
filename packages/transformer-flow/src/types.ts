import t from '@babel/types';
import { Node, Visitor } from '@babel/traverse';

export interface Warning {
  info: string;
  issueUrl: string;
  node: t.Node;
}

type RuleA = () => Visitor<Node>;
type RuleB = (warnings: Warning[]) => Visitor<Node>;
export type Rule = RuleA | RuleB;
