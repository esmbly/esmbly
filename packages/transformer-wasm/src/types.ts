export interface WasmTransformerOptions {
  optimize?: string;
  optimizeLevel?: number;
  shrinkLevel?: number;
  validate?: boolean;
  // TODO: ADD more options
}
