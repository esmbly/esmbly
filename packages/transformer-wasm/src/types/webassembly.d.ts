/*
WebAssembly v1 (MVP) declaration file for TypeScript
Definitions by: 01alchemist (https://twitter.com/01alchemist)

MIT License

Copyright (c) 2017 01Alchemist

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
SOFTWARE.
*/

declare namespace WebAssembly {
  /**
   * WebAssembly.Module
   **/
  class Module {
      constructor (bufferSource: ArrayBuffer | Uint8Array);

      static customSections(module: Module, sectionName: string): ArrayBuffer[];
      static exports(module: Module): {
          name: string;
          kind: string;
      }[];
      static imports(module: Module): {
          module: string;
          name: string;
          kind: string;
      }[];
  }

  /**
   * WebAssembly.Instance
   **/
  class Instance {
      readonly exports: any;
      constructor (module: Module, importObject?: any);
  }

  /**
   * WebAssembly.Memory
   * Note: A WebAssembly page has a constant size of 65,536 bytes, i.e., 64KiB.
   **/
  interface MemoryDescriptor {
      initial: number;
      maximum?: number;
  }

  class Memory {
      readonly buffer: ArrayBuffer;
      constructor (memoryDescriptor: MemoryDescriptor);
      grow(numPages: number): number;
  }

  /**
   * WebAssembly.Table
   **/
  interface TableDescriptor {
      element: "anyfunc",
      initial: number;
      maximum?: number;
  }

  class Table {
      readonly length: number;
      constructor (tableDescriptor: TableDescriptor);
      get(index: number): Function;
      grow(numElements: number): number;
      set(index: number, value: Function): void;
  }

  /**
   * Errors
   */
  class CompileError extends Error {
      readonly fileName: string;
      readonly lineNumber: string;
      readonly columnNumber: string;
      constructor (message?:string, fileName?:string, lineNumber?:number);
      toString(): string;
  }

  class LinkError extends Error {
      readonly fileName: string;
      readonly lineNumber: string;
      readonly columnNumber: string;
      constructor (message?:string, fileName?:string, lineNumber?:number);
      toString(): string;
  }

  class RuntimeError extends Error {
      readonly fileName: string;
      readonly lineNumber: string;
      readonly columnNumber: string;
      constructor (message?:string, fileName?:string, lineNumber?:number);
      toString(): string;
  }

  function compile(bufferSource: ArrayBuffer | Uint8Array): Promise<Module>;

  interface ResultObject {
      module: Module;
      instance: Instance;
  }

  function instantiate(bufferSource: ArrayBuffer | Uint8Array, importObject?: any): Promise<ResultObject>;
  function instantiate(module: Module, importObject?: any): Promise<Instance>;

  function validate(bufferSource: ArrayBuffer | Uint8Array): boolean;
}