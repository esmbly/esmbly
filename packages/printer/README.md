# `@esmbly/printer`
<p>
  <a title="MIT License" href="LICENSE">
    <img src="https://img.shields.io/github/license/gridsome/gridsome.svg?style=flat-square&label=License&colorB=6cc24a">
  </a>
  <a title="Build Status" href="https://travis-ci.org/esmbly/esmbly">
    <img src="https://travis-ci.org/esmbly/esmbly.svg?branch=master">
  </a>
  <a href="https://coveralls.io/github/esmbly/esmbly?branch=master" href="Coverage Status">
    <img src="https://coveralls.io/repos/github/esmbly/esmbly/badge.svg?branch=master" />
  </a>
</p>

Shared printing utilities for **Esmbly**.

Used by `@esmbly/core`, `@esmbly/cli` and the transformers. 

## Usage
```js
import { printer } from '@esmbly/printer';

// Print some info
printer.print('some info..');

// Print an error
printer.error('an error..');

// Print warnings
printer.printWarnings([Warning]);

// Silence the printer
printer.setErrorStream(null);
printer.setOutStream(null);
```

## Contributing
All types of contributions are very much welcome. Check out our [**Contributing Guide**](CONTRIBUTING.md) for instructions on how to get started.
