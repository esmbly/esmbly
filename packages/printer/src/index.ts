import { Warning } from '@esmbly/types';
import chalk from 'chalk';
import { Callback, Encoding, WriteFn } from './types';
import * as templates from './templates';

class Printer {
  private stdout: NodeJS.WriteStream | null;
  private stderr: NodeJS.WriteStream | null;

  public constructor() {
    this.stdout = process ? process.stdout : null;
    this.stderr = process ? process.stderr : null;
  }

  public get isTTY(): boolean {
    if (this.stdout) {
      return this.stdout.isTTY === true;
    }

    return false;
  }

  public error(
    chunk: string,
    encoding?: Encoding,
    callback?: Callback,
  ): boolean {
    if (this.stderr) {
      return this.stderr.write(chunk, encoding, callback);
    }

    return false;
  }

  public print(
    chunk: string,
    encoding?: Encoding,
    callback?: Callback,
  ): boolean {
    if (this.stdout) {
      return this.stdout.write(chunk, encoding, callback);
    }

    return false;
  }

  public printWarnings(warnings: Warning[]): void {
    if (this.stdout) {
      warnings.forEach((warning: Warning) => {
        this.print(templates.warning(warning));
      });
    }
  }

  public forceEnableColors(): void {
    chalk.enabled = true;
  }

  public forceDisableColors(): void {
    chalk.enabled = false;
  }

  public clearLine(...args: WriteFn[]): void {
    if (this.isTTY) {
      args.forEach(fn => fn.call(this, '\x1b[999D\x1b[K'));
    }
  }

  public setErrorStream(
    stream: NodeJS.WriteStream | NodeJS.WritableStream | null,
  ): void {
    this.stderr = stream as NodeJS.WriteStream;
  }

  public setOutStream(
    stream: NodeJS.WriteStream | NodeJS.WritableStream | null,
  ): void {
    this.stdout = stream as NodeJS.WriteStream;
  }
}

export const printer = new Printer();
export { templates };
