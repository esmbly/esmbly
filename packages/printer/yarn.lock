# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
# yarn lockfile v1


"@babel/code-frame@^7.12.13":
  version "7.12.13"
  resolved "https://registry.yarnpkg.com/@babel/code-frame/-/code-frame-7.12.13.tgz#dcfc826beef65e75c50e21d3837d7d95798dd658"
  integrity sha512-HV1Cm0Q3ZrpCR93tkWOYiuYIgLxZXZFVG2VgK+MBWjUqZTundupbfx2aXarXuw5Ko5aMcjtJgbSs4vUGBS5v6g==
  dependencies:
    "@babel/highlight" "^7.12.13"

"@babel/generator@^7.14.0":
  version "7.14.1"
  resolved "https://registry.yarnpkg.com/@babel/generator/-/generator-7.14.1.tgz#1f99331babd65700183628da186f36f63d615c93"
  integrity sha512-TMGhsXMXCP/O1WtQmZjpEYDhCYC9vFhayWZPJSZCGkPJgUqX0rF0wwtrYvnzVxIjcF80tkUertXVk5cwqi5cAQ==
  dependencies:
    "@babel/types" "^7.14.1"
    jsesc "^2.5.1"
    source-map "^0.5.0"

"@babel/helper-function-name@^7.12.13":
  version "7.12.13"
  resolved "https://registry.yarnpkg.com/@babel/helper-function-name/-/helper-function-name-7.12.13.tgz#93ad656db3c3c2232559fd7b2c3dbdcbe0eb377a"
  integrity sha512-TZvmPn0UOqmvi5G4vvw0qZTpVptGkB1GL61R6lKvrSdIxGm5Pky7Q3fpKiIkQCAtRCBUwB0PaThlx9vebCDSwA==
  dependencies:
    "@babel/helper-get-function-arity" "^7.12.13"
    "@babel/template" "^7.12.13"
    "@babel/types" "^7.12.13"

"@babel/helper-get-function-arity@^7.12.13":
  version "7.12.13"
  resolved "https://registry.yarnpkg.com/@babel/helper-get-function-arity/-/helper-get-function-arity-7.12.13.tgz#bc63451d403a3b3082b97e1d8b3fe5bd4091e583"
  integrity sha512-DjEVzQNz5LICkzN0REdpD5prGoidvbdYk1BVgRUOINaWJP2t6avB27X1guXK1kXNrX0WMfsrm1A/ZBthYuIMQg==
  dependencies:
    "@babel/types" "^7.12.13"

"@babel/helper-split-export-declaration@^7.12.13":
  version "7.12.13"
  resolved "https://registry.yarnpkg.com/@babel/helper-split-export-declaration/-/helper-split-export-declaration-7.12.13.tgz#e9430be00baf3e88b0e13e6f9d4eaf2136372b05"
  integrity sha512-tCJDltF83htUtXx5NLcaDqRmknv652ZWCHyoTETf1CXYJdPC7nohZohjUgieXhv0hTJdRf2FjDueFehdNucpzg==
  dependencies:
    "@babel/types" "^7.12.13"

"@babel/helper-validator-identifier@^7.14.0":
  version "7.14.0"
  resolved "https://registry.yarnpkg.com/@babel/helper-validator-identifier/-/helper-validator-identifier-7.14.0.tgz#d26cad8a47c65286b15df1547319a5d0bcf27288"
  integrity sha512-V3ts7zMSu5lfiwWDVWzRDGIN+lnCEUdaXgtVHJgLb1rGaA6jMrtB9EmE7L18foXJIE8Un/A/h6NJfGQp/e1J4A==

"@babel/highlight@^7.12.13":
  version "7.14.0"
  resolved "https://registry.yarnpkg.com/@babel/highlight/-/highlight-7.14.0.tgz#3197e375711ef6bf834e67d0daec88e4f46113cf"
  integrity sha512-YSCOwxvTYEIMSGaBQb5kDDsCopDdiUGsqpatp3fOlI4+2HQSkTmEVWnVuySdAC5EWCqSWWTv0ib63RjR7dTBdg==
  dependencies:
    "@babel/helper-validator-identifier" "^7.14.0"
    chalk "^2.0.0"
    js-tokens "^4.0.0"

"@babel/parser@^7.12.13", "@babel/parser@^7.14.0", "@babel/parser@^7.4.3":
  version "7.14.1"
  resolved "https://registry.yarnpkg.com/@babel/parser/-/parser-7.14.1.tgz#1bd644b5db3f5797c4479d89ec1817fe02b84c47"
  integrity sha512-muUGEKu8E/ftMTPlNp+mc6zL3E9zKWmF5sDHZ5MSsoTP9Wyz64AhEf9kD08xYJ7w6Hdcu8H550ircnPyWSIF0Q==

"@babel/template@^7.12.13":
  version "7.12.13"
  resolved "https://registry.yarnpkg.com/@babel/template/-/template-7.12.13.tgz#530265be8a2589dbb37523844c5bcb55947fb327"
  integrity sha512-/7xxiGA57xMo/P2GVvdEumr8ONhFOhfgq2ihK3h1e6THqzTAkHbkXgB0xI9yeTfIUoH3+oAeHhqm/I43OTbbjA==
  dependencies:
    "@babel/code-frame" "^7.12.13"
    "@babel/parser" "^7.12.13"
    "@babel/types" "^7.12.13"

"@babel/traverse@^7.4.3":
  version "7.14.0"
  resolved "https://registry.yarnpkg.com/@babel/traverse/-/traverse-7.14.0.tgz#cea0dc8ae7e2b1dec65f512f39f3483e8cc95aef"
  integrity sha512-dZ/a371EE5XNhTHomvtuLTUyx6UEoJmYX+DT5zBCQN3McHemsuIaKKYqsc/fs26BEkHs/lBZy0J571LP5z9kQA==
  dependencies:
    "@babel/code-frame" "^7.12.13"
    "@babel/generator" "^7.14.0"
    "@babel/helper-function-name" "^7.12.13"
    "@babel/helper-split-export-declaration" "^7.12.13"
    "@babel/parser" "^7.14.0"
    "@babel/types" "^7.14.0"
    debug "^4.1.0"
    globals "^11.1.0"

"@babel/types@^7.12.13", "@babel/types@^7.14.0", "@babel/types@^7.14.1", "@babel/types@^7.3.4":
  version "7.14.1"
  resolved "https://registry.yarnpkg.com/@babel/types/-/types-7.14.1.tgz#095bd12f1c08ab63eff6e8f7745fa7c9cc15a9db"
  integrity sha512-S13Qe85fzLs3gYRUnrpyeIrBJIMYv33qSTg1qoBwiG6nPKwUWAD9odSzWhEedpwOIzSEI6gbdQIWEMiCI42iBA==
  dependencies:
    "@babel/helper-validator-identifier" "^7.14.0"
    to-fast-properties "^2.0.0"

"@babel/types@^7.4.4":
  version "7.4.4"
  resolved "https://registry.yarnpkg.com/@babel/types/-/types-7.4.4.tgz#8db9e9a629bb7c29370009b4b779ed93fe57d5f0"
  integrity sha512-dOllgYdnEFOebhkKCjzSVFqw/PmmB8pH6RGOWkY4GsboQNd47b1fBThBSwlHAq9alF9vc1M3+6oqR47R50L0tQ==
  dependencies:
    esutils "^2.0.2"
    lodash "^4.17.11"
    to-fast-properties "^2.0.0"

"@esmbly/types@^0.0.6":
  version "0.0.6"
  resolved "https://registry.yarnpkg.com/@esmbly/types/-/types-0.0.6.tgz#abcebbab6a0cfe7ebd1e92102c60fad457dcb690"
  integrity sha512-pRt+JfiZ3w4LBtKfgjchuJQgN2uhfUjWHHjtdCVNxkB6BCQ0jvVMxxtmr21Yiw1mihp4a+ZOp1/rkU7sZNx+SQ==
  dependencies:
    "@babel/parser" "^7.4.3"
    "@babel/traverse" "^7.4.3"
    "@babel/types" "^7.3.4"

ansi-styles@^3.2.1:
  version "3.2.1"
  resolved "https://registry.yarnpkg.com/ansi-styles/-/ansi-styles-3.2.1.tgz#41fbb20243e50b12be0f04b8dedbf07520ce841d"
  integrity sha512-VT0ZI6kZRdTh8YyJw3SMbYm/u+NqfsAxEpWO0Pf9sq8/e94WxxOpPKx9FR1FlyCtOVDNOQ+8ntlqFxiRc+r5qA==
  dependencies:
    color-convert "^1.9.0"

chalk@^2.0.0, chalk@^2.4.2:
  version "2.4.2"
  resolved "https://registry.yarnpkg.com/chalk/-/chalk-2.4.2.tgz#cd42541677a54333cf541a49108c1432b44c9424"
  integrity sha512-Mti+f9lpJNcwF4tWV8/OrTTtF1gZi+f8FqlyAdouralcFWFQWF2+NgCHShjkCb+IFBLq9buZwE1xckQU4peSuQ==
  dependencies:
    ansi-styles "^3.2.1"
    escape-string-regexp "^1.0.5"
    supports-color "^5.3.0"

color-convert@^1.9.0:
  version "1.9.3"
  resolved "https://registry.yarnpkg.com/color-convert/-/color-convert-1.9.3.tgz#bb71850690e1f136567de629d2d5471deda4c1e8"
  integrity sha512-QfAUtd+vFdAtFQcC8CCyYt1fYWxSqAiK2cSD6zDB8N3cpsEBAvRxp9zOGg6G/SHHJYAT88/az/IuDGALsNVbGg==
  dependencies:
    color-name "1.1.3"

color-name@1.1.3:
  version "1.1.3"
  resolved "https://registry.yarnpkg.com/color-name/-/color-name-1.1.3.tgz#a7d0558bd89c42f795dd42328f740831ca53bc25"
  integrity sha1-p9BVi9icQveV3UIyj3QIMcpTvCU=

debug@^4.1.0:
  version "4.3.1"
  resolved "https://registry.yarnpkg.com/debug/-/debug-4.3.1.tgz#f0d229c505e0c6d8c49ac553d1b13dc183f6b2ee"
  integrity sha512-doEwdvm4PCeK4K3RQN2ZC2BYUBaxwLARCqZmMjtF8a51J2Rb0xpVloFRnCODwqjpwnAoao4pelN8l3RJdv3gRQ==
  dependencies:
    ms "2.1.2"

escape-string-regexp@^1.0.5:
  version "1.0.5"
  resolved "https://registry.yarnpkg.com/escape-string-regexp/-/escape-string-regexp-1.0.5.tgz#1b61c0562190a8dff6ae3bb2cf0200ca130b86d4"
  integrity sha1-G2HAViGQqN/2rjuyzwIAyhMLhtQ=

esutils@^2.0.2:
  version "2.0.2"
  resolved "https://registry.yarnpkg.com/esutils/-/esutils-2.0.2.tgz#0abf4f1caa5bcb1f7a9d8acc6dea4faaa04bac9b"
  integrity sha1-Cr9PHKpbyx96nYrMbepPqqBLrJs=

globals@^11.1.0:
  version "11.12.0"
  resolved "https://registry.yarnpkg.com/globals/-/globals-11.12.0.tgz#ab8795338868a0babd8525758018c2a7eb95c42e"
  integrity sha512-WOBp/EEGUiIsJSp7wcv/y6MO+lV9UoncWqxuFfm8eBwzWNgyfBd6Gz+IeKQ9jCmyhoH99g15M3T+QaVHFjizVA==

has-flag@^3.0.0:
  version "3.0.0"
  resolved "https://registry.yarnpkg.com/has-flag/-/has-flag-3.0.0.tgz#b5d454dc2199ae225699f3467e5a07f3b955bafd"
  integrity sha1-tdRU3CGZriJWmfNGfloH87lVuv0=

js-tokens@^4.0.0:
  version "4.0.0"
  resolved "https://registry.yarnpkg.com/js-tokens/-/js-tokens-4.0.0.tgz#19203fb59991df98e3a287050d4647cdeaf32499"
  integrity sha512-RdJUflcE3cUzKiMqQgsCu06FPu9UdIJO0beYbPhHN4k6apgJtifcoCtT9bcxOpYBtpD2kCM6Sbzg4CausW/PKQ==

jsesc@^2.5.1:
  version "2.5.2"
  resolved "https://registry.yarnpkg.com/jsesc/-/jsesc-2.5.2.tgz#80564d2e483dacf6e8ef209650a67df3f0c283a4"
  integrity sha512-OYu7XEzjkCQ3C5Ps3QIZsQfNpqoJyZZA99wd9aWd05NCtC5pWOkShK2mkL6HXQR6/Cy2lbNdPlZBpuQHXE63gA==

lodash@^4.17.11:
  version "4.17.21"
  resolved "https://registry.yarnpkg.com/lodash/-/lodash-4.17.21.tgz#679591c564c3bffaae8454cf0b3df370c3d6911c"
  integrity sha512-v2kDEe57lecTulaDIuNTPy3Ry4gLGJ6Z1O3vE1krgXZNrsQ+LFTGHVxVjcXPs17LhbZVGedAJv8XZ1tvj5FvSg==

ms@2.1.2:
  version "2.1.2"
  resolved "https://registry.yarnpkg.com/ms/-/ms-2.1.2.tgz#d09d1f357b443f493382a8eb3ccd183872ae6009"
  integrity sha512-sGkPx+VjMtmA6MX27oA4FBFELFCZZ4S4XqeGOXCv68tT+jb3vk/RyaKWP0PTKyWtmLSM0b+adUTEvbs1PEaH2w==

source-map@^0.5.0:
  version "0.5.7"
  resolved "https://registry.yarnpkg.com/source-map/-/source-map-0.5.7.tgz#8a039d2d1021d22d1ea14c80d8ea468ba2ef3fcc"
  integrity sha1-igOdLRAh0i0eoUyA2OpGi6LvP8w=

supports-color@^5.3.0:
  version "5.5.0"
  resolved "https://registry.yarnpkg.com/supports-color/-/supports-color-5.5.0.tgz#e2e69a44ac8772f78a1ec0b35b689df6530efc8f"
  integrity sha512-QjVjwdXIt408MIiAqCX4oUKsgU2EqAGzs2Ppkm4aQYbjm+ZEWEcW4SfFNTr4uMNZma0ey4f5lgLrkB0aX0QMow==
  dependencies:
    has-flag "^3.0.0"

to-fast-properties@^2.0.0:
  version "2.0.0"
  resolved "https://registry.yarnpkg.com/to-fast-properties/-/to-fast-properties-2.0.0.tgz#dc5e698cbd079265bc73e0377681a4e4e83f616e"
  integrity sha1-3F5pjL0HkmW8c+A3doGk5Og/YW4=
