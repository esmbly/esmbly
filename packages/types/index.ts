export * from './AST';
export * from './File';
export * from './FileType';
export * from './RunConfig';
export * from './Transformer';
