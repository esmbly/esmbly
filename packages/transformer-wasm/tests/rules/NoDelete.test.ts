import { Writable } from 'stream';
import printer from '@esmbly/printer';
import { Format } from '@esmbly/types';
import testRunner from '../helpers/testRunner';

const program = `
  function noDelete(a: number[]): number[] {
    if (a.length > 0) {
      delete a[1];
    }
    return a;
  }
`;

describe('rule: NoDelete', () => {
  beforeAll(() => {
    printer.setOutStream(new Writable({ write: () => {} }));
    printer.forceDisableColors();
  });

  afterAll(() => {
    printer.setOutStream(process.stdout);
    printer.forceEnableColors();
  });

  it('throws an error that warns about the use of delete', async () => {
    const output = [{ format: Format.AssemblyScript }];
    await expect(
      testRunner(program, output),
    ).rejects.toThrowErrorMatchingSnapshot();
  });
});
