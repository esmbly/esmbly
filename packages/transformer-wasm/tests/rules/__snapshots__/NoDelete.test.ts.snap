// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rule: NoDelete throws an error that warns about the use of delete 1`] = `
"
  [33mwarning[39m [2mAssemblyScript does not support the delete operator[22m

    [31mdelete a[1];[39m
  
  [2mat:[22m [36mtestfile.ts:4:6[39m
  [2missue:[22m [36mhttps://github.com/AssemblyScript/assemblyscript/wiki/Limitations[39m

"
`;
